fix: add `--preserve-env=PATH` to `sudo` command

dicej opened the pull request:
Previously, run_servers.sh would check that all the necessary prerequisites are
in the current shell's PATH, but since most of them are actually run via sudo,
what we really care about is whether those prerequisites are in *root's* PATH.
Now we check root's PATH for those things before proceeding.

This is important because the user may have installed Spin or Bindle using
`cargo install`, which only installs things for the current user, not system-wide.

Signed-off-by: Joel Dice <joel.dice@gmail.com>

bacongobbler commented on run_servers.sh:
```diff
@@ -10,11 +10,22 @@ require() {
+
+# The following will be run directly as the current user, and thus must appear in the current PATH:
 require vault
+require consul
```
slight nitpick here, but can we put this above line 27 so the `require` lines are all sorted alphabetically?

bacongobbler commented on run_servers.sh:
```diff
@@ -10,11 +10,22 @@ require() {
+# The following will be run (either directly or indirectly) via sudo, and thus must appear in the root user's PATH:
+require_root bindle-server
+require_root nomad
+require_root traefik
```
same here on alphabetical order - can we move this below `spin`?

bacongobbler reviewed (approved):
LGTM! Just had a few comments on alphabetical sorting. Not necessary to address before merging.

dicej commented:
Good call -- they're now alphabetized.

bacongobbler reviewed (approved):
double LGTM - I'll let @adamreese merge this one as he's the primary maintainer of this project.

bacongobbler commented:
No. This is required for production. Nomad must run as root in Linux for the `exec` driver to work.

https://www.nomadproject.io/docs/drivers/exec#client-requirements

Because `nomad` is executed as root, any jobs it spawns runs as root, so the binaries for `bindle`, `traefik`, and `spin` all require to be in the root user's PATH.

bacongobbler commented:
For MacOS users, this is irrelevant because `raw_exec` does not require to be run as root.

https://www.nomadproject.io/docs/drivers/raw_exec#client-requirements

adamreese commented:
In what cases would these not be in sudo's path?

bacongobbler commented:
> This is important because the user may have installed Spin or Bindle using
> `cargo install`, which only installs things for the current user, not system-wide.

bacongobbler commented:
I commonly place dev tools in `$HOME/bin`. In those cases, these binaries would not be accessible by root's `$PATH`.

dicej committed:
fix: add `--preserve-env=PATH` to `sudo` command

This ensures that Nomad inherits the current user's `PATH` environment variable,
rather that the default `sudo` behavior of using a sanitized PATH.

This is necessary because the user may have installed `bindle-server`,
`traefik`, etc. locally but not system-wide.  For example, `cargo install`
installs in the current user's home directory by default.

Signed-off-by: Joel Dice <joel.dice@gmail.com>

dicej commented:
Just force-pushed a different approach:

> fix: add `--preserve-env=PATH` to `sudo` command
    
> This ensures that Nomad inherits the current user's `PATH` environment variable,
> rather that the default `sudo` behavior of using a sanitized PATH.
    
> This is necessary because the user may have installed `bindle-server`,
> `traefik`, etc. locally but not system-wide.  For example, `cargo install`
> installs in the current user's home directory by default.
